src/GitVersionTask/IGitVersionTaskExecutor.cs

```diff
@@ -0,0 +1,12 @@
+using GitVersionTask.MsBuild.Tasks;
+
+namespace GitVersionTask
+{
+    public interface IGitVersionTaskExecutor
```

Comment:
I somehow feel there should be some kind of relationship between `IGitVersionExecutor` and `IGitVersionTaskExecutor`. 🤔

Reply:
They are both entrypoint to the GitVersion calculation. The IGitVersionExecutor takes the command line inputs and convert to arguments and later it calls the calculation logic. IGitVersionTaskExecutor takes the msbuild ITask as input, and converts that into arguments and later it calls the calculation logic. They are similar, and probably we'll extract a common Executor, but let's leave it for another PR. It's already a huge PR. Let's move this to feature/di3 branch